aws/data_source_aws_route53_resolver_rule.go

```diff
@@ -122,7 +122,9 @@ func dataSourceAwsRoute53ResolverRuleRead(d *schema.ResourceData, meta interface
 	d.SetId(aws.StringValue(rule.Id))
 	arn := *rule.Arn
 	d.Set("arn", arn)
-	d.Set("domain_name", rule.DomainName)
+	// To be consistent with other AWS services that do not accept a trailing period,
+	// we remove the suffix from the Domain Name returned from the API
+	d.Set("domain_name", trimTrailingPeriod(aws.StringValue(rule.DomainName)))
```

Comment:
Are you seeing the result of the API include a trailing period? The [`ListResolverRules` API Reference example](https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_ListResolverRules.html) does not show one (and we should prefer to do nothing special in the resource code where possible 😄 ). If not, I'd recommend leaving the existing logic.

Reply:
Note for self/readers: we're keeping this logic here as the `rule.DomainName` value returned from the API comes back with a trailing period, even when `domain_name` is not explicitly configured